lib/buffer.js

```diff
@@ -340,9 +340,14 @@ function slowToString(encoding, start, end) {
 
 Buffer.prototype.toString = function() {
   const length = this.length | 0;
-  if (arguments.length === 0)
-    return this.utf8Slice(0, length);
-  return slowToString.apply(this, arguments);
+  if (arguments.length === 0) {
+    var result = this.utf8Slice(0, length);
```

Comment:
Seeing how this is the only branch that uses `length`, you might as well fold it into the call, i.e. `return utf8Slice(0, this.length)`.